### PR TITLE
Migrate to Bevy 0.15

### PR DESCRIPTION
**Pull request by haihala**
Hey so I don't know what I'm doing. Followed rust-analyzer, clippy, the tests and what little intuition I have, I got it to compile and the examples to run. I have a wonky setup and can't be a 100% sure even the box game works, but the synctest seems to, so I'm relying on that.

For the love of everything good and holy, please do read through this and see if you can make it better. Especially the places where `bevy::utils::get_short_name` were used, as I never figured out if that was necessary and/or how to do it now that get_short_name no longer exists. Maybe [this link to the migration guide](https://bevyengine.org/learn/migration-guides/0-14-to-0-15/#remove-allocation-in-get-short-name) helps, but the `ShortName` type doesn't seem to exist.

### CODE REVIEW

File: examples/stress_tests/particles.rs [hunk 1]
```diff
@@ -170,7 +170,6 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         app.rollback_component_with_reflect::<Sprite>()
             .rollback_component_with_reflect::<Transform>()
             .rollback_component_with_reflect::<GlobalTransform>()
-            .rollback_component_with_reflect::<Handle<Image>>()
```
Comment (johanhelsing) [inline examples/stress_tests/particles.rs:173]: Handles as component were removed in 0.15, but where did this particular handle go?

File: src/snapshot/component_checksum.rs [hunk 2]
```diff
@@ -88,7 +88,7 @@ where
 
             trace!(
                 "Component {} has checksum {:X}",
-                bevy::utils::get_short_name(std::any::type_name::<C>()),
+                std::any::type_name::<C>(),
```
Comment (johanhelsing) [inline src/snapshot/component_checksum.rs:91]: Yeah, like you said in the pr body, we should figure out what to do here :)

File: examples/stress_tests/particles.rs [hunk 3]
```diff
@@ -139,7 +139,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     let mut session_builder = SessionBuilder::<Config>::new()
         .with_num_players(num_players)
         .with_desync_detection_mode(desync_mode)
-        .with_max_prediction_window(args.max_prediction)?
```
Comment (johanhelsing) [inline examples/stress_tests/particles.rs:142]: I guess this became non-optional in ggrs? Seems unrelated to the bevy upgrade, not necessarily a problem, just wanted to check.

**Review by johanhelsing — COMMENTED** [review]
Thanks for porting! We should clarify what happened to get_short_name, and if that image handle is moved somewhere else that we should rollback, but otherwise looks good :)

File: examples/stress_tests/particles.rs (hunk 1 above)
Comment (haihala) [reply examples/stress_tests/particles.rs:173]: In Sprite

File: src/snapshot/component_checksum.rs (hunk 2 above)
Comment (haihala) [reply src/snapshot/component_checksum.rs:91]: Relevant discord conversations: can be found [here](https://discord.com/channels/691052431525675048/1002362493634629796/1314525820877803591)

File: examples/stress_tests/particles.rs (hunk 3 above)
Comment (haihala) [reply examples/stress_tests/particles.rs:142]: Yeah, but the way the ggrs dependency is set up I think I upgraded that by acciden.

File: src/snapshot/component_checksum.rs (hunk 2 above)
Comment (haihala) [reply src/snapshot/component_checksum.rs:91]: Sages of the discord have said we should use disqualified, since it was created for this. Will do so after I get home.

**Comment by haihala** [comment]
I consulted several discord channels, and they claim that the disqualified crate was split off from Bevy and we should use it here. I then did just that.

**Review by johanhelsing — APPROVED** [review]
Looks good to me! This pr was a lot cleaner than i expected reading the description. Thanks for doing all the digging after short names.

**Comment by haihala** [comment]
> Looks good to me! This pr was a lot cleaner than i expected reading the description. Thanks for doing all the digging after short names.

Thanks for the kind words. I don't have much expertise with contributing to open source and don't know the crates, so I was guessing a log. Really speaks to the quality of rust tools that I even got there in the end.

I don't appear to have rights to merge and I'm almost certain the pipeline doesn't deploy to crates.io. Do you know how to proceed?

**Comment by johanhelsing** [comment]
@gschup publishes manually. I have merge rights, but I'll wait a bit for him, and maybe his review, since he needs to do the publishing anyway.

**Comment by johanhelsing** [comment]
I think this will need some extra changes when https://github.com/gschup/ggrs/pull/82 is merged

**Comment by johanhelsing** [comment]
@haihala I pushed some fixes for latest ggrs changes in #114
